hbase-client/src/main/java/org/apache/hadoop/hbase/filter/FilterListWithAND.java

```diff
@@ -184,9 +205,20 @@ public ReturnCode filterCell(Cell c) throws IOException {
       // otherwise we may mess up the global state (such as offset, count..) in the following
       // sub-filters. (HBASE-20565)
       if (!isIncludeRelatedReturnCode(rc)) {
-        return rc;
+        i++;
```

Comment:
Ditto.

I think the intention here is that, usually SEEK_NEXT_USING_HINT can skip more rows, so even if we decide to skip the current cell, we'd better still looking to remaining filters to see if we can get a SEEK_NEXT_USING_HINT.

Reply:
Yes, exactly.

Reply:
will add more comments.

Reply:
Let's speak more here? It is not easy for a developer without the context of this issue here to understand, and please also mention the issue number.